[Feat] 추가문의 컴포넌트 구현

vcz-Chan opened the pull request:
### 💡 To Reviewers


- 확인을 위해 상세 페이지에 넣었습니다 확인 시 http://localhost:3000/inquiries/1 로 가주시면 되요!
- api명세를 기반으로 목테이터, 타입 작성했습니다.
- 입력창 위에 파란색으로 강조된 수신자 정보를 표시하였습니다.
  - 입력 내용이 길어져 스크롤이 발생할 경우, 수신자 정보(파란 영역)와 겹치는 문제가 있어, 스크롤이 발생하면 해당 영역이 사라지도록 처리했습니다.
  - 수신자 이름은 사람마다 길이가 다르므로, prefix 형태로 관리하였습니다.
  - 이 prefix는 입력창 내부에 함께 들어가 간격을 맞추지만 위의 수신자 정보에 가려저 보이지는 않습니다, 또한 사용자가 수정하거나 삭제할 수 없도록 고정되어 있습니다.
    - 입력값 전송 시 prefix는 제외하고, 본문만 서버에 전달할 예정입니다.
    
- 댓글하고 대 댓글에서 수신자 정보가 없습니다. 
  - api명세서를 조율 해야 할 것 같아 아무거나 넣어놨습니다.
  - 수정도 위의 수신자 정보를 받아야 기존 값을 바탕으로 수정 폼을 띄울 수 있을 것 같습니다.

- 입력한 내용과 수신자 선택을 해야만 완료 버튼이 활성화 되게 했습니다.

- 리뷰 반영해 수정 완료했습니다.


---

### 🔥 작업 내용

- 추가문의 헤더
- 추가문의 본문
- 추가문의 입력
- 댓글, 대댓글 입력
- 댓글 대댓글 본문
- 수정 기능 구현

---

### 🤔 추후 작업 예정

- api 명세 변경 후 수정, 수신자 로직 보완

---

### 📸 작업 결과 (선택)

기본 페이지
<img width="1219" height="628" alt="image" src="https://github.com/user-attachments/assets/45d10585-affc-4218-9583-966ce8cee29a" />

추가문의 작성
<img width="1201" height="425" alt="image" src="https://github.com/user-attachments/assets/6e24ba9d-0787-47e8-bc4a-4bd9efae4842" />

추가문의 작성 - 수신자 선택
<img width="1217" height="361" alt="image" src="https://github.com/user-attachments/assets/d8a50238-3d71-4acd-94f7-79f68297300a" />

추가문의 작성 - 글 작성 -> 버튼 활성화
<img width="1195" height="365" alt="image" src="https://github.com/user-attachments/assets/ed735181-7350-4725-b400-7fd956592587" />

추가문의 수정
<img width="1137" height="464" alt="image" src="https://github.com/user-attachments/assets/91a8f416-344d-42ee-96eb-644de61f631a" />

댓글, 대댓글
<img width="1190" height="599" alt="image" src="https://github.com/user-attachments/assets/d541e3a4-5309-460e-b758-f127c39b2291" />

댓글 작성
<img width="1187" height="500" alt="스크린샷 2025-07-31 오전 11 48 23" src="https://github.com/user-attachments/assets/d27a0db4-778e-46bb-b493-ea85206a804c" />

댓글 수정
<img width="1172" height="394" alt="image" src="https://github.com/user-attachments/assets/6b60c8e8-9db7-4f02-97b5-996523bb61d1" />


---

### 🔗 관련 이슈 (선택)

- #27

vcz-Chan committed:
feat: 추가문의 컴포넌트 구현

- 추가문의 헤더
- 추가문의 본문
- 추가문의 입력
- 댓글, 대댓글 입력
- 댓글 대댓글 본문

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryThread.tsx:
```diff
@@ -0,0 +1,88 @@
+          {/* 2. 대댓글(하위) */}
+          <div className="py-8 px-6 rounded-[15px] bg-gray-10 flex flex-col gap-8 ">
+            {fu.comments.map(c => (
+              <div className="flex gap-4 ">
```
<img width="285" height="86" alt="스크린샷 2025-07-30 오후 3 33 17" src="https://github.com/user-attachments/assets/10368489-b716-4ea4-9b2e-2883b84b583b" />

key 속성 넣어주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryHeader.tsx:
```diff
@@ -0,0 +1,34 @@
+    <div className="w-full  h-16 flex justify-between items-center">
+      <div className="flex gap-2 items-center">
+        <span className="text-heading2-sb">추가 문의 </span>
+        <span className="py-[2px] px-3 rounded-[30px] w-[35px] h-[26px] bg-main text-white text-body1 text-center">
```
고정 넓이로 하면 cnt 많아질수록 넘쳐날 수 있으니 w-fit으로 변경해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryHeader.tsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <div className="w-full  h-16 flex justify-between items-center">
+      <div className="flex gap-2 items-center">
+        <span className="text-heading2-sb">추가 문의 </span>
```
text-gray-100도 명시해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryHeader.tsx:
```diff
@@ -0,0 +1,34 @@
+          onClick={onClick}
+          className="w-16 h-16 rounded-[15px] bg-main px-6 text-white cursor-pointer"
+        >
+          <Pen />
```
이미 있는 svg 활용하시면 좋을 거 같아여
import Pen from "@/assets/svgs/inquiry/pencil.svg";
에 있는데 common 폴더로 빼주시고 사용해주세요
추가한 svg는 제거!

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryReplyForm.tsx:
```diff
@@ -0,0 +1,82 @@
+import { Pen } from "@/assets/svgs/AdditionalInquiry";
+
+export type Props = {
+  parenWriter: string; // 부모 글쓴이 이름
```
parentWriter로 고고

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryBody.tsx:
```diff
@@ -0,0 +1,83 @@
+    const hh = String(d.getHours()).padStart(2, "0");
+    const mm = String(d.getMinutes()).padStart(2, "0");
+    return `${YYYY}년 ${MM}월 ${DD}일 ${hh}:${mm}`;
+  };
```
utils 폴더에 분리해주세요!

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryBody.tsx:
```diff
@@ -0,0 +1,83 @@
+  };
+
+  return (
+    <div className="w-full flex flex-col gap-8">
```
gap-4로 수정해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryBody.tsx:
```diff
@@ -0,0 +1,83 @@
+      </div>
+
+      {/* 본문 */}
+      <div className="text-body2 text-gray-800 whitespace-pre-wrap mb-4">
```
위에서 gap-4넣었으니 mb-4 빼셔도 될 거 같아요

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryHeader.tsx:
```diff
@@ -0,0 +1,34 @@
+  onClick,
+}: Props) => {
+  return (
+    <div className="w-full  h-16 flex justify-between items-center">
```
펜 버튼이 있을 경우에는 items-center, 버튼을 눌러서 사라졌을 때는 이 부분이 items-start 가 되어야 할 거 같슴다

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryReplyForm.tsx:
```diff
@@ -0,0 +1,82 @@
+            : "bg-gray-20 text-gray-80 cursor-not-allowed"
+        }`}
+      >
+        <Pen className="w-4 h-4" />
```
이 부분도 기존에 있는 이미지로 사용해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryForm.tsx:
```diff
@@ -0,0 +1,104 @@
+  return (
+    <div className="w-full rounded-[15px] p-8 border-3 border-main flex gap-8 ">
+      <div className="flex-1 min-h-[150px] flex flex-col gap-8">
+        <div className="flex gap-5 items-center">
```
<img width="621" height="127" alt="스크린샷 2025-07-30 오후 4 07 54" src="https://github.com/user-attachments/assets/af677904-158c-4fc7-985d-17f2a6509bc7" />

문의 대상끼리 한번 묶어주시고 gap-4 적용해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryForm.tsx:
textarea 구현하실 때 사용자가 입력하는 높이만큼 인풋창 높이가 커질 수 있도록 
```
const textareaRef = useRef<HTMLTextAreaElement>(null);
```
추가해주시고 

handleChange 내부에

```

   if (textareaRef.current) {
      textareaRef.current.style.height = "auto"; 
      textareaRef.current.style.height =
        textareaRef.current.scrollHeight + "px";
    }
```

추가해주세요!! 최대 높이가 있는지는 슬랙에 남겨놔서 답변오면 코드 남겨둘게요

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryReplyForm.tsx:
여기도 textarea 높이 신경써주세요!!!

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryThread.tsx:
```diff
@@ -0,0 +1,88 @@
+
+          {/* 1-1) 추가문의 답변 폼 */}
+          {replyToId === fu.follow_up_id && (
+            <div className="">
```
이 div 태그는 추후 사용하실 예정이 아니라면 삭제해주세용

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryThread.tsx:
```diff
@@ -0,0 +1,88 @@
+                  {/* 대댓글 답변 폼 */}
+                  {replyToId === c.comment_id && (
+                    <AdditionalInquiryReplyForm
+                      parenWriter={fu.writer.name}
```
여기 {c.writer.name}가 들어가야하는게 아닐까요?-?

chaeyoungwon commented on src/types/InquiryResponse.ts:
굳

chaeyoungwon commented on src/components/AdditionalInquiry/AdditionalInquiryReplyForm.tsx:
```diff
@@ -0,0 +1,82 @@
+        </div>
+      </div>
+
+      {/* 완료 버튼 */}
```
이런 버튼들은 공통 버튼 컴포넌트 사용가능하다면 사용해주세요!!
/common/Button.tsx 에 정의되어있슴다

chaeyoungwon reviewed (commented):
꽤나 복잡한데 정말 수고 많으셨습니다
prefix 넣으시는 거 성공하셨군요 👍 👍

vcz-Chan committed:
refactor: 추가 문의 컴포넌트 리펙토링

- 변수명 변경
- 디자인 요소 수정
- 입력창 크기 늘어나게 변경
- 키값 누락 수정
- 유틸 함수 외부로 이동
- 공통 아이콘관리
- 공통 버튼 컴포넌트로 변경

vcz-Chan committed:
feat: 추가 문의 및 댓글/대댓글 수정 기능 구현

- 추가 문의, 댓글, 대댓글 수정 시 수정 폼이 표시되도록 구현
- 수정 시 초기값으로 기존 내용을 반영하는 기능은 API 명세 수정 후 작업 예정

vcz-Chan committed:
refactor: 추가 문의 컴포넌트 리펙토링

- api연동을 위한 주석 추가
- 변수명, 함수명 수정

Kuuuuu00 commented on src/assets/svgs/commons/pencil.svg:
엇 commons 폴더 지웠는데.. svgs/layout에서 import 해주세요

Kuuuuu00 commented on src/pages/inquiryDetail/InquiryDetailPage.tsx:
```diff
@@ -1,7 +1,10 @@
+
 const InquiryDetailPage = () => {
   return (
     <section className="w-full h-[835px] bg-gray-50">
```
bg-gray-50 빼주세용

Kuuuuu00 commented on src/components/AdditionalInquiry/AdditionalInquiryHeader.tsx:
```diff
@@ -0,0 +1,37 @@
+        <Button
+          buttonType="blue"
+          onClick={onClick}
+          className="w-16 h-16 rounded-[15px] p-0"
```
p-0 없어도 될 거 같아요

Kuuuuu00 commented on src/components/AdditionalInquiry/AdditionalInquiryThread.tsx:
```diff
@@ -0,0 +1,127 @@
+            </div>
+          )}
+
+          {/* 2. 댓글(하위) */}
+          <div className="py-8 px-6 rounded-[15px] bg-gray-10 flex flex-col gap-8 ">
+            {fu.comments.map(c => (
+              <div className="flex gap-4" key={c.comment_id}>
+                <Enter />
+                <div className="flex-1 flex flex-col gap-8">
+                  {editCommentId === c.comment_id ? (
+                    <AdditionalInquiryReplyForm
+                      recipient={fu.writer.name} //여기를 댓글 대댓글의 수신자로 변경 c.recipient or c.recipient.name
+                      onClose={handleFormClose}
+                      // todo: onSubmit={댓글 수정}
+                      // initialContent={c.content} // 초기 내용 전달
+                      // initialRecipient={c.recipient.name} // 초기 수신자 전달
+                    />
+                  ) : (
+                    <AdditionalInquiryBody
+                      recipient={fu.writer.name} //여기를 댓글 대댓글의 수신자로 변경 c.recipient or c.recipient.name
+                      writer={c.writer}
+                      created_at={c.created_at}
+                      content={c.content}
+                      canEdit={true} // 수정 여부 c.writer.user_id === 로그인한 사용자 ID
+                      onAnswer={() => handleAnswerOpen(c.comment_id)}
+                      onEdit={() => handleEditComment(c.comment_id)}
+                    />
+                  )}
+                  {/* 대댓글 답변 폼 */}
+                  {replyToId === c.comment_id && (
+                    <AdditionalInquiryReplyForm
+                      recipient={c.writer.name}
+                      onClose={handleFormClose}
+                      // todo: onSubmit={대댓글 추가}
+                    />
+                  )}
+                </div>
+              </div>
+            ))}
+          </div>
```
```suggestion
         <div className="py-8 px-6 rounded-[15px] bg-gray-10 flex flex-col">
            {fu.comments.map(c => (
              <div key={c.comment_id}>
                <div className="flex gap-4 mb-8">
                  <Enter />
                  <div className="flex-1 flex flex-col gap-8">
                    {editCommentId === c.comment_id ? (
                      <AdditionalInquiryReplyForm
                        recipient={fu.writer.name} //여기를 댓글 대댓글의 수신자로 변경 c.recipient or c.recipient.name
                        onClose={handleFormClose}
                        // todo: onSubmit={댓글 수정}
                        // initialContent={c.content} // 초기 내용 전달
                        // initialRecipient={c.recipient.name} // 초기 수신자 전달
                      />
                    ) : (
                      <AdditionalInquiryBody
                        recipient={fu.writer.name} //여기를 댓글 대댓글의 수신자로 변경 c.recipient or c.recipient.name
                        writer={c.writer}
                        created_at={c.created_at}
                        content={c.content}
                        canEdit={true} // 수정 여부 c.writer.user_id === 로그인한 사용자 ID
                        onAnswer={() => handleAnswerOpen(c.comment_id)}
                        onEdit={() => handleEditComment(c.comment_id)}
                      />
                    )}
                  </div>
                </div>
                {/* 대댓글 답변 폼 */}
                {replyToId === c.comment_id && (
                  <AdditionalInquiryReplyForm
                    recipient={c.writer.name}
                    onClose={handleFormClose}
                    // todo: onSubmit={대댓글 추가}
                  />
                )}
              </div>
            ))}
          </div>
```
<img width="1512" height="943" alt="image" src="https://github.com/user-attachments/assets/a9fe1f37-5b72-4e83-aacb-26ba206fed2e" />

디자인 상 대댓글 답변 폼이 왼쪽 끝에 붙어서 시작하는거 같습니다!
이렇게 Enter를 map 안으로 옮겨주시고 

관련된 gap 수정해주시면 될 거 같습니다

Kuuuuu00 reviewed (commented):
신경쓸게 되게 많은 작업이었을 거 같은데 고생많으셨습니다!
리뷰 반영해서 다시 푸쉬 부탁드립니다~

vcz-Chan committed:
feat: 추가 문의 및 대댓글 답변, 수정 폼 개선

- 답변 시 입력 폼을 디자인에 맞게 수정
- 수정 시 입력 폼을 디자인에 맞게 조정
- SVG 아이콘 위치 수정

chaeyoungwon committed:
merge: dev 충돌 수정

letthem commented on src/assets/svgs/AdditionalInquiry/UserProfile.svg:
이 Profile 이미지도 common 에 있는 거 쓰시면 될 것 같습니다!

letthem commented on src/components/AdditionalInquiry/AdditionalInquiryBody.tsx:
```diff
@@ -0,0 +1,73 @@
+            alt={writer.name}
+            className="w-5 h-5 rounded-full"
+          /> */}
+          <UserProfile className="w-5 h-5 rounded-full" />
```
여기는 기본으로 writer.profile_image_url 에서 받아오되, writer.profile_image_url 이 null 이면 UserProfile 띄우게 로직 바꾸는 게 더 좋을 것 같습니다!

letthem commented on src/components/AdditionalInquiry/AdditionalInquiryForm.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  const handleScroll = (e: React.UIEvent<HTMLTextAreaElement>) => {
+    setHidePrefix(e.currentTarget.scrollTop > 0);
+  };
```
handleChange에서 textarea 높이 늘어나게 처리해서 이 handleScroll 로직 및 hidePrefix state는 없애도 됩니다!

letthem commented on src/components/AdditionalInquiry/AdditionalInquiryReplyForm.tsx:
```diff
@@ -0,0 +1,83 @@
+
+  const handleScroll = (e: React.UIEvent<HTMLTextAreaElement>) => {
+    setHidePrefix(e.currentTarget.scrollTop > 0);
+  };
```
여기도 삭제!

letthem reviewed (commented):
고생많으셨습니다 👍🍀😊

vcz-Chan committed:
refactor: 프로필 이미지 로직 개선 및 스크롤 처리 제거

- writer.profile_image_url 유무에 따라 기본 이미지(UserProfile) 또는 실제 이미지 표시
- 목데이터에 프로필 이미지 추가
- textarea의 높이를 자동으로 확장되도록 수정
- 높이 제한이 없어짐에 따라 handleScroll 및 hidePrefix 관련 로직 제거

chaeyoungwon committed:
Merge branch 'dev' into feat/27-additionalInquiry-component
